docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb

```diff
@@ -0,0 +1,530 @@
+    "\n",
+    "### Understanding Model Signatures\n",
+    "\n",
+    "A model signature in MLflow describes the schema for inputs, outputs, and parameters of an ML model. It is a blueprint that details the expected data types and shapes, facilitating a clear interface for model usage. Signatures are particularly useful as they are:\n",
+    "\n",
+    "- Displayed in MLflow's UI for easy reference.\n",
+    "- Employed by MLflow's deployment tools to validate inputs during inference.\n",
+    "- Stored in a standardized JSON format alongside the model's metadata.\n",
+    "\n",
+    "### The Role of Signatures in Code\n",
+    "\n",
+    "In the following section, we are using MLflow to infer the signature of a machine learning model. This involves specifying an input example, generating a model output example, and defining any additional inference parameters. The resulting signature is used to validate future inputs and document the expected data formats.\n",
+    "\n",
+    "### Types of Model Signatures\n",
+    "\n",
+    "Model signatures can be:\n",
+    "\n",
+    "- **Column-based**: Suitable for models that operate on tabular data, with each column having a specified data type and optional name.\n",
+    "- **Tensor-based**: Designed for models that take tensors as inputs and outputs, with each tensor having a specified data type, shape, and optional name.\n",
+    "- **With Params**: Some models require additional parameters for inference, which can also be included in the signature.\n",
+    "\n",
+    "For the transformers flavor, all input types are of the Column-based type (referred to within MLflow as `ColSpec` types).\n",
+    "\n",
+    "### Signature Enforcement\n",
+    "\n",
+    "MLflow enforces the signature at the time of model inference, ensuring that the provided input and parameters match the expected schema. If there's a mismatch, MLflow will raise an exception or issue a warning, depending on the nature of the mismatch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/benjamin.wilson/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/transformers/generation/utils.py:1260: UserWarning: Using the model-agnostic default `max_length` (=20) to control the generation length. We recommend setting `max_new_tokens` to control the maximum length of the generation.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "signature = mlflow.models.infer_signature(\n",
+    "    input_example,\n",
+    "    mlflow.transformers.generate_signature_output(generation_pipeline, input_example),\n",
+    "    parameters,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting the tracking server\n",
```

Comment:
Can we remove this section? For demonstrating purposes, I think we can just use the default local file store.

Reply:
+1, it seems the need of a server (which is not true but perceptionally) is a psychological hurdle to start MLflow

Reply:
> Explain how to connect to hosted MLflow (Databricks CE)

Can we explain this in a different section?